glean-core/src/lib.rs

```diff
@@ -181,7 +179,7 @@ impl Glean {
         let event_data_store = EventDatabase::new(&cfg.data_path)?;
 
         let mut glean = Self {
-            upload_enabled: cfg.upload_enabled,
+            upload_enabled: false,
```

Comment:
nit: let's add a comment to not scare our future selves out. This _should not_ trigger a wipeout of the stored client id.